src/components/Auth/UserFormDialog/UserForm.tsx

```diff
@@ -1,85 +1,145 @@
+  authUserDialogData,
+  clearAuthUserDialogData,
+  updateUser,
+  createUser,
   user,
-  onSave,
-  onClose,
-  isEditing,
 }) => {
+  const isEditing = !!user;
+
   const form = useForm<AddAuthUserPayload>({
     defaultValues: user,
     mode: 'onChange',
   });
+
+  const save = useCallback(
+    (newUser: AddAuthUserPayload) => {
+      if (isEditing) {
+        updateUser({ user: newUser, localId: user?.localId! });
+      } else {
+        createUser({ user: newUser });
+      }
+    },
+    [isEditing, updateUser, createUser, user]
+  );
+
   const { register, handleSubmit, formState, reset, errors } = form;
-  const canSubmit =
-    formState.isValid && Object.values(formState.touched).length > 0;
+
+  const canSubmit = !authUserDialogData?.loading && formState.isValid;
 
   const submit = useCallback(
     (user: AddAuthUserPayload) => {
       // Take into account multi-field errors.
       if (Object.values(errors).length === 0) {
-        onSave(user);
-        onClose();
+        save(user);
+        clearAuthUserDialogData();
       }
     },
-    [errors, onSave, onClose]
+    [errors, save, clearAuthUserDialogData]
   );
 
   return (
-    <form onSubmit={handleSubmit(submit)} data-testid="user-form">
-      <DialogContent>
-        <Field
-          name="displayName"
-          label="Display name (optional)"
-          type="text"
-          placeholder="Enter display name"
-          error={errors?.displayName && 'Display name is required'}
-          inputRef={register({})}
-        />
+    <Dialog renderToPortal open onClose={clearAuthUserDialogData}>
+      <DialogTitle>
+        {user ? `Edit User ${user.displayName}` : 'Add a user'}
+      </DialogTitle>
+
+      <form onSubmit={handleSubmit(submit)} data-testid="user-form">
+        <DialogContent>
+          <Field
+            name="displayName"
+            label="Display name (optional)"
+            type="text"
+            placeholder="Enter display name"
+            error={errors?.displayName && 'Display name is required'}
+            inputRef={register({})}
+          />
 
-        <ImageUrlInput {...form} />
-        <CustomAttributes {...form} />
-        <SignInMethod {...form} user={user}></SignInMethod>
-      </DialogContent>
-      <DialogActions>
-        <DialogButton action="close" type="button" theme="secondary">
-          Cancel
-        </DialogButton>
-        {!isEditing && (
-          <Button
-            onClick={handleSubmit(result => {
-              onSave(result);
-              reset(user);
-            })}
-            disabled={!canSubmit}
-            type="button"
-          >
-            Save and create another
-          </Button>
-        )}
+          <ImageUrlInput {...form} />
+          <CustomAttributes {...form} />
+          <SignInMethod {...form} user={user} />
+          {hasError(authUserDialogData?.result) && (
+            <Callout type="warning">Unknown server error</Callout>
```

Comment:
Have we considered surfacing the error? It will make debugging easier (or at least make future GitHub issues less cryptic)

Reply:
Going to show the error message coming from the server